app/controllers/api/base_controller/renderer.rb

```diff
@@ -252,11 +252,11 @@ def split_virtual_attribute(attr)
       end
 
       def attr_accessible?(object, attr)
-        return false unless object && object.respond_to?(attr)
-        object.class.has_attribute?(attr) ||
-          object.class.reflect_on_association(attr) ||
-          object.class.virtual_attribute?(attr) ||
-          object.class.virtual_reflection?(attr)
+        return true if object && object.respond_to?(attr)
+        (object.class.respond_to?(:has_attribute?) && object.class.has_attribute?(attr)) ||
```

Comment:
I was following the pattern from attr_virtual? method below.  I guess we could "try" something like:

```
object.class.try(:has_attribute?, attr)
  || object.class.try(:virtual_attribute?, attr)
  || object.class.try(:virtual_reflection?, attr)
```

should work too.  I'm good either way.